src/lib/operators/table_scan/abstract_dereferenced_column_table_scan_impl.cpp

```diff
@@ -71,6 +76,21 @@ void AbstractDereferencedColumnTableScanImpl::_scan_reference_segment(const Refe
       matches[match_idx].chunk_offset = sub_pos_list.original_positions[matches[match_idx].chunk_offset];
     }
   }
+
+  if (predicate_condition != PredicateCondition::IsNull) {
+    return;
+  }
+
+  // For PredicateCondition::IsNull, split_pos_list_by_chunk_id() stores all NULL_ROW_IDs in an extra SubPosList at the
+  // end of chunk_offsets_by_chunk_id. These are then retrieved and written to the matches.
+
+  const auto& sub_pos_list = chunk_offsets_by_chunk_id[referenced_chunk_count];
+  const auto& remaining_null_row_id_positions = *sub_pos_list.row_ids;
+
+  matches.reserve(matches.size() + remaining_null_row_id_positions.size());
+  for (const auto& pos : remaining_null_row_id_positions) {
+    matches.emplace_back(chunk_id, pos.chunk_offset);
```

Comment:
Now that I see it ...

Since we know the sizes upfront, maybe use `resize()` and `[]` to write the positions?